app/routes.py

```diff
@@ -1,2 +1,333 @@
-from flask import Blueprint
+import re
+from flask import Blueprint, jsonify, request
+from flask.helpers import make_response
+from app.models.task import Task
+from app.models.goal import Goal
+import requests
+from datetime import datetime 
+from app import db
+import os
+from dotenv import load_dotenv
 
+tasks_bp = Blueprint("tasks_bp", __name__, url_prefix="/tasks")
+
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    request_body = request.get_json()
+    
+    if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
+        return jsonify({"details": "Invalid data"}), 400
+    
+    new_task = Task(
+        title=request_body["title"], 
+        description=request_body["description"], 
+        completed_at=request_body["completed_at"]
+        )
+    
+    db.session.add(new_task) 
+    db.session.commit()
+    
+    response_body = {
+        "task": {
+            "id": new_task.task_id,
+            "title": new_task.title,
+            "description": new_task.description,
+            "is_complete": new_task.completed_at is not None
+            }
+        }
+    
+    return jsonify(response_body), 201
+
+@tasks_bp.route("", methods=["GET"])
+def read_tasks():
+    title_query = request.args.get("sort")
+
+    if title_query == "asc":
+        tasks = Task.query.order_by(Task.title.asc())
+    elif title_query == "desc":
+        tasks = Task.query.order_by(Task.title.desc())
+    else:
+        tasks = Task.query.all() 
+    
+    task_responses = []
+    for task in tasks:
+        task_responses.append(
+                {
+                "id": task.task_id,
+                "title": task.title,
+                "description": task.description,
+                "is_complete": task.completed_at is not None
+                }
+            )
+    
+    return jsonify(task_responses), 200
+
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def read_task(task_id):
+    task_id = int(task_id)
+    task = Task.query.get(task_id)
+    
+    if task == None:
+        return jsonify(None), 404
+
+    if not task.goal_id:
+        return jsonify({
+            "task": {
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False if task.completed_at == None else True
+        }
+        }), 200
+    else:
+        return jsonify({
+            "task": {
+            "id": task.task_id,
+            "goal_id": task.goal_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete": False if task.completed_at == None else True
+        }
+        }), 200
```

Comment:
Notice how these conditionals return a dictionary with the same key value pairs (besides goal_id) ? Instead of explicitly writing out this same dictionary twice, we can write a "base" dictionary and add the goal_id to it if it expects for the task.

```suggestion
    task_dict = {
            "id": task.task_id,
            "title": task.title,
            "description": task.description,
            "is_complete": False if task.completed_at == None else True
        }

    if task.goal_id:
        task_dict["goal_id] = task.goal_id   
    return task_dict, 200
```

Reply:
We can also have this same logic within a Task class helper method as well!